runhouse/resources/hardware/kubernetes_cluster.py

```diff
@@ -0,0 +1,127 @@
+import sky
+from pathlib import Path
+import subprocess
+import logging
+import time
+
+from runhouse.resources.hardware.on_demand_cluster import OnDemandCluster
+
+logger = logging.getLogger(__name__)
+
+class KubernetesCluster(OnDemandCluster):
+
+    def up(self):
+        """Up the cluster.
+
+        Example:
+            >>> rh.kubernetes_cluster(
+            >>>     name="cpu-cluster-test",
+            >>>     instance_type="CPU:1",
+            >>>     provider="kubernetes",      
+            >>> )
+        """
+        if self.on_this_cluster():
+            return self
+
+        if self.provider in ["kubernetes"]:
```

Comment:
should we just default the provider for `kubernetes_cluster()` to "kubernetes" and throw an error in the factory function if they input something else?

Reply:
Yeah, I think that would be good.